### PR TITLE
Improve error message when pushing/appending with promote=true

### PR DESCRIPTION
**Pull request by bkamins**
Improve error in cases like:

```
append!(DataFrame(x=PooledArray(1:255, UInt8)), DataFrame(x=PooledArray(256:500, UInt8)); promote=true)
```

as reported by @jariji

### CODE REVIEW

File: test/insertion.jl [hunk 1]
```diff
@@ -1343,4 +1343,31 @@ end
...
+
+    with_logger(sl) do
+        @test_throws ErrorException append!(DataFrame(x=PooledArray(1:255, UInt8)),
+                                            DataFrame(x=PooledArray(256:500, UInt8)))
+        @test_throws ErrorException append!(DataFrame(x=PooledArray(1:255, UInt8)),
+                                            DataFrame(x=PooledArray(256:500, UInt8)),
+                                            promote=true)
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), [1000])
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), [1000],
+                                        promote=true)
```
Comment (nalimilan) [inline test/insertion.jl:1358]:
```suggestion
        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), [1000],
                                          promote=true)
```

File: test/insertion.jl [hunk 2]
```diff
@@ -1343,4 +1343,31 @@ end
...
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
+                                        promote=true)
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
+                                        cols=:union, promote=true)
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
+                                        cols=:union, promote=false)
+        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x="x",),
+                                        cols=:union, promote=true)
+        @test_throws MethodError push!(DataFrame(x=PooledArray(1:255, UInt8)), (x="x",),
+                                    cols=:union, promote=false)
```
Comment (nalimilan) [inline test/insertion.jl:1369]:
```suggestion
        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
                                          promote=true)
        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
                                          cols=:union, promote=true)
        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x=1000,),
                                          cols=:union, promote=false)
        @test_throws ErrorException push!(DataFrame(x=PooledArray(1:255, UInt8)), (x="x",),
                                          cols=:union, promote=true)
        @test_throws MethodError push!(DataFrame(x=PooledArray(1:255, UInt8)), (x="x",),
                                       cols=:union, promote=false)
```

File: src/dataframe/insertion.jl [hunk 3]
```diff
@@ -297,10 +297,14 @@ function _append_or_prepend!(df1::DataFrame, df2::AbstractDataFrame; cols::Symbo
...
-               "Maybe it was forgotten to require column " *
-               "element type promotion, which can be done " *
-               "by passing the promote=true keyword argument."
+        if promote
+            @error "Error adding value to column :$(_names(df1)[current_col])."
+        else
+            @error "Error adding value to column :$(_names(df1)[current_col]). " *
+                "Maybe it was forgotten to require column " *
+                "element type promotion, which can be done " *
+                "by passing the promote=true keyword argument."
```
Comment (nalimilan) [inline src/dataframe/insertion.jl:306]: While we're at it, maybe it would be enough to say "Maybe you forgot passing `promote=true`?"?

**Review by nalimilan — APPROVED** [review]

File: src/dataframe/insertion.jl (hunk 3 above)
Comment (bkamins) [reply src/dataframe/insertion.jl:306]: OK - changed.

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**